en/1/04-structs.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Structs
+actions: ['checkAnswer', 'hints']
+material:
+  editor:
+    language: rust
+    startingCode: |
+      #![no_std]
+
+      multiversx_sc::imports!();
+      multiversx_sc::derive_imports!();
+
+      // start here
+
+      #[multiversx_sc::contract]
+      pub trait ZombiesContract {
+
+        #[init]
+        fn init(&self) {
+          self.dna_digits().set(16u8);
+        }
+
+        #[storage_mapper("dna_digits")]
+        fn dna_digits(&self) -> SingleValueMapper<u8>;
+      }
+    answer: >
+      #![no_std]
+
+      multiversx_sc::imports!();
+      multiversx_sc::derive_imports!();
+
+      #[derive(NestedEncode, NestedDecode, TopEncode, TopDecode, TypeAbi)]
+      pub struct Zombie<M: ManagedTypeApi> {
+          name: ManagedBuffer<M>,
+          dna: u64,
+      }
+
+      #[multiversx_sc::contract]
+      pub trait ZombiesContract {
+
+        #[init]
+        fn init(&self) {
+          self.dna_digits().set(16u8);
+        }
+
+        #[storage_mapper("dna_digits")]
+        fn dna_digits(&self) -> SingleValueMapper<u8>;
+      }
+---
+
+# The dynamic allocation problem
+
+Many basic Rust types (like String and Vec<T>) are dynamically allocated on the heap. In simple terms, it means the program (in this case, the smart contract) keeps asking for more and more memory from the runtime environment (the VM). For small collections, this doesn't matter much, but for bigger collection, this can become slow and the VM might even stop the contract and mark the execution as failed.
+
+The managed types work by only storing a handle within the contract memory, which is a u32 index, while the actual payload resides in reserved VM memory. More information about this can be found in **MultiversX**'s documentation https://docs.multiversx.com/developers/best-practices/the-dynamic-allocation-problem/
```

Comment:
- What is a handle?
- After reading this paragraph, I feel like I don't grasp the new concepts that have been introduced, and I feel more confused than before the paragraph. I don't think putting the link to the documentation would help as at this moment I don't think I would want quitting the tutorial to read another one.